files/image_config/reboot-cause/process-reboot-cause

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# process-reboot-cause
+#
+# Program designed to read the previous reboot-cause files, log the last previous reboot-cause.
+# And read the saved reboot-cause history files and save the reboot cause in the state-db.
+#
+
+try:
+    import json
+    import os
+    import pwd
+    import re
+    import sys
+
+    import swsssdk
+    from sonic_py_common import device_info, logger
+except ImportError as err:
+    raise ImportError("%s - required module not found" % str(err))
+
+VERSION = "1.0"
+
+SYSLOG_IDENTIFIER = "process-reboot-cause"
+
+REBOOT_CAUSE_DIR = "/host/reboot-cause/"
+PREVIOUS_REBOOT_CAUSE_DIR = "/host/reboot-cause/previous-reboot-cause/"
+PREVIOUS_REBOOT_CAUSE_FILE = REBOOT_CAUSE_DIR + "last-reboot-cause.json"
```

Comment:
Prefer sticking with "previous-" rather than "last-", as it's a bit more specific, whereas "last" can have a few meanings.